### DISCUSSION TITLE
utils::createInstance() quick question

### DISCUSSION THREAD

**Discussion by rudiHammad** [axmolengine/axmol#1045] [Q&A] [answered]
Hi,
beeing new to axmol, I am looking at some differences with cocos2dx 4. I' ve had an overview at the list of changes.
I just have a question about utils::createInstance().
In cocos2dx we used to add CREATE_FUNC in the header to run the init() and call the auto release method so we don't forget it.
In axmol, in the AppDelegate I noticed that we use auto scene = utils::createInstance<HelloWorld>(), (and no need of CREATE_FUNC).
Looking at the definition documentation it says  _Create a Game Object with 'bool T::init()', like CREATE_FUNC, but more powerful_

I guess my question is how is it more powerful? I am not sure I understand with a bit more of explanation.

thanks,
R

**Comment by halx99** [comment, marked as answer]
The purpose is don't require CREATE_FUNC
